platform.md

```diff
@@ -395,22 +415,34 @@ The lifecycle:
 - SHALL write the resolved build plan from the detected group to `<plan>`
 - SHALL provide `run-image.target` data in `<analyzed>` to buildpacks according to the process outlined in the [Buildpack Interface Specification](buildpack.md).
 
-When image extensions are present in the order (**[experimental](#experimental-features)**), the lifecycle:
+When image extensions are present in the order (optional and **[experimental](#experimental-features)**), the lifecycle:
 - SHALL execute all image extensions in the order defined in `<group>` according to the process outlined in the [Buildpack Interface Specification](buildpack.md).
+- SHALL filter the build plan with dependencies provided by image extensions.
 - SHALL copy any generated run.Dockerfiles to `<generated>/run/<image extension ID>/Dockerfile`.
 - SHALL copy any generated build.Dockerfiles to `<generated>/build/<image extension ID>/Dockerfile`.
 - SHALL copy any generated `<extend-config>` files to `<generated>/build/<image extension ID>/<extend-config>`.
-- SHALL replace the `run-image` reference in `<analyzed>` with the selected run image reference. The selected run image reference SHALL be the base image referenced in the Dockerfile output by the last image extension in the group.
-- SHALL filter the build plan with dependencies provided by image extensions.
+- SHALL replace `run-image` in `<analyzed>` with the selected run image. To select the run image, the lifecycle SHALL inspect each `run.Dockerfile` output by image extensions, in the order defined in `<group>`:
+  - **If** all `run.Dockerfile`s declare `FROM ${base_image}`, the selected run image SHALL be the original run image in `<analyzed>`, with `extend = true`
+  - **Else** the selected run image SHALL be the last image referenced in the `FROM` statement of the last `run.Dockerfile` not to declare `FROM ${base_image}`
+    - `run-image.image` SHALL be the name of the selected run image
+    - `run-image.reference` and `run-image.target` SHALL be cleared (as they may no longer be accurate)
+    - All preceding `run.Dockerfile`s SHALL be renamed `run.Dockerfile.ignore`
+    - **If** there are no `run.Dockerfile`s following the Dockerfile with the selected run image:
+      - `run-image.extend` SHALL be `false`
+    - **Else**
+      - `run-image.extend` SHALL be `true`
+- SHALL fail if the selected run image is not found in `<run>`
```

Comment:
This may be nicer if it doesn't fail outright, but is a warning, or at least could be reduced to a warning via a flag or env var. 

I'm thinking here that it may be required in some circumstances for customers to override the selected run image to one of their choosing (buildpacks supports this today for non-switched images, and arguably an extension may be able to make a more nuanced choice).. 

When combined with stack removal, which may make it a lot easier to use an image as a run image, it does make sense to have some way to override, although there are various options.. 

 - force override to a single given image
 - force override via a map of run.toml image -> desired image
 - force override to a constructed image reference using some rule applied via extensions (eg, baseimage:extension1-extension2-extension3)
 
 The 2nd & 3rd options although probably most powerful in conjunction with extension image switching also feel potentially too big to be containable within this item, but either way, would require the weakening of the 'SHALL' within the above statement.

Reply:
Changed to a warning